### PR TITLE
chore: Signup Page Component tests

### PR DESCRIPTION
**Pull request by tharun208**
**What this PR does / why we need it**: This PR introduces a test using the `rtl` library for `SignupPage` and `VerifyEmailPage`.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
1. Added new aria labels for the signup page.
2. Added proper validation for an email in the `VerifyEmail` component.

### CODE REVIEW

**Comment by tharun208** [comment]
@Clarity-89 can I get a review on this?

**Comment by Clarity-89** [comment]
> @Clarity-89 can I get a review on this?

You don't need to explicitly mention anyone on the PR to get a review, assigning as a reviewer usually is enough. We have an internal hackaton this week and some of the team members are on holidays so it might take longer than usual to get reviews on some PRs.

File: public/app/core/components/Signup/SignupPage.tsx [hunk 1]
```diff
@@ -63,10 +64,11 @@ export const SignupPage: FC<Props> = (props) => {
           {({ errors, register, getValues }: FormAPI<SignupDTO>) => (
             <>
               <Field label="Your name">
-                <Input {...register('name')} placeholder="(optional)" />
+                <Input aria-label={selectors.pages.Signup.name} {...register('name')} placeholder="(optional)" />
```
Comment (Clarity-89) [inline public/app/core/components/Signup/SignupPage.tsx:67]:
```suggestion
                <Input {...register('name')} id="user_name" placeholder="(optional)" />
```
Is there a reason for adding the aria labels here? I think adding `id` would make the inputs accessible enough, as then the label will be correctly associated with the inputs (our `Field` component adds `htmlFor` attribute [automatically](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Forms/Field.tsx#L79-L88) for the input elements with `id`). We mostly use aria-labels in e2e tests and for components that do not have accessible role or name. This applies to all the other inputs in this component.

**Review by Clarity-89 — COMMENTED** [review]
Looks good! Left one comment to consider.

File: public/app/core/components/Signup/SignupPage.tsx (hunk 1 above)
Comment (tharun208) [reply public/app/core/components/Signup/SignupPage.tsx:67]: sure, will do the cleanup and use `id` where ever i added `aria-label`

Comment (tharun208) [reply public/app/core/components/Signup/SignupPage.tsx]: addressed

File: public/app/core/components/Signup/SignupPage.tsx [hunk 2]
```diff
@@ -80,16 +81,17 @@ export const SignupPage: FC<Props> = (props) => {
               </Field>
               {!getConfig().autoAssignOrg && (
                 <Field label="Org. name">
-                  <Input {...register('orgName')} placeholder="Org. name" />
+                  <Input id="org_name" {...register('orgName')} placeholder="Org. name" />
```
Comment (Clarity-89) [inline public/app/core/components/Signup/SignupPage.tsx:84]:
```suggestion
                  <Input id="org-name" {...register('orgName')} placeholder="Org. name" />
```
I think we'd have a consistent naming pattern for ids.

File: public/app/core/components/Signup/SignupPage.tsx [hunk 3]
```diff
@@ -108,7 +111,9 @@ export const SignupPage: FC<Props> = (props) => {
               </Field>
 
               <HorizontalGroup>
-                <Button type="submit">Submit</Button>
+                <Button type="submit" id="submit">
```
Comment (Clarity-89) [inline public/app/core/components/Signup/SignupPage.tsx:114]:
```suggestion
                <Button type="submit">
```
ID is not necessary for buttons.

**Review by Clarity-89 — COMMENTED** [review]
Almost there, just a few tiny suggestions.

File: public/app/core/components/Signup/SignupPage.tsx (hunk 2 above)
Comment (tharun208) [reply public/app/core/components/Signup/SignupPage.tsx]: addressed

File: public/app/core/components/Signup/SignupPage.tsx (hunk 3 above)
Comment (tharun208) [reply public/app/core/components/Signup/SignupPage.tsx]: addressed

**Review by Clarity-89 — APPROVED** [review]
Great work! Thanks for adding the tests 🎉

**Pull request merged**